src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2450,11 +2454,15 @@ export class StudyViewPageStore {
     readonly initialVisibleAttributesClinicalDataBinCountData = remoteData<DataBin[]>({
         await: () => [this.defaultVisibleAttributes],
         invoke: async () => {
-            const
-                    attributes= _.uniqBy( _.filter(this.defaultVisibleAttributes.result, attr => attr.datatype === 'NUMBER').map(attr => {
-                        return this._clinicalDataBinFilterSet.get(getClinicalAttributeUniqueKey( attr))!;
-                    }),attr => `${attr.attributeId}_${attr.clinicalDataType}`
-            );
+            const attributes = _.chain(this.defaultVisibleAttributes.result)
+                .filter(attr => attr.datatype === 'NUMBER')
+                .map(attr => ({
```

Comment:
Could you move the initialization of `ClinicalDataBinFilter` to a function so that the same logic in `readonly clinicalAttributes = remoteData({` could be used?